src/FastCholesky.jl

```diff
@@ -35,17 +35,10 @@ end
 fastcholesky(input::Diagonal) = cholesky(input)
 fastcholesky(input::Hermitian) = cholesky(PositiveFactorizations.Positive, input)
 fastcholesky(input::Number) = input
+fastcholesky!(input::Number) = input
 
-function fastcholesky(x::UniformScaling)
-    return error(
-        "`fastcholesky` is not defined for `UniformScaling`. The shape is not determined."
-    )
-end
-function fastcholesky!(x::UniformScaling)
-    return error(
-        "`fastcholesky!` is not defined for `UniformScaling`. The shape is not determined."
-    )
-end
+fastcholesky(x::UniformScaling) = sqrt(x.λ) * I
```

Comment:
I think this operation should throw, because this behaviour is not compatible with the output of the other types. `fastcholesky` must return a Cholesky factorization in the form of the `Cholesky` structure, but here it simply returns a squared-root of the matrix. For this operation we have a different method called `cholsqrt`.